x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts

```diff
@@ -0,0 +1,391 @@
+  static incrementalIdMissingFilter = 'not cases.attributes.incremental_id: *';
+  private isStopped = false;
+
+  constructor(
+    private internalSavedObjectsClient: SavedObjectsClientContract,
+    private logger: Logger
+  ) {
+    this.logger = logger.get('incremental_id_service');
+    this.logger.debug('Cases incremental ID service initialized');
+  }
+
+  public stopService() {
+    this.isStopped = true;
+  }
+  public startService() {
+    this.isStopped = false;
+  }
+
+  public async getCasesWithoutIncrementalId(parameters: Omit<GetCasesParameters, 'filter'> = {}) {
+    return this.getCases({
+      ...parameters,
+      filter: CasesIncrementalIdService.incrementalIdMissingFilter,
+    });
+  }
+
+  public async getCases({
+    filter,
+    perPage = 10000,
+    page = 1,
+    sortOrder = 'asc',
+    sortField = 'created_at',
+    namespaces = ['*'],
+  }: GetCasesParameters) {
+    try {
+      const savedCases = await this.internalSavedObjectsClient.find<CasePersistedAttributes>({
+        type: CASE_SAVED_OBJECT,
+        sortField,
+        sortOrder,
+        perPage,
+        page,
+        filter,
+        namespaces,
+      });
+      return savedCases;
+    } catch (error) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  /**
+   * Get the latest applied ID for a given space.
+   * Uses the actually applied numerical ids on cases in the space.
+   */
+  public async getLastAppliedIdForSpace(namespace: string) {
+    try {
+      const casesResponse = await this.getCases({
+        filter: CasesIncrementalIdService.incrementalIdExistsFilter,
+        namespaces: [namespace],
+        sortField: 'incremental_id',
+        sortOrder: 'desc',
+        perPage: 1, // We only need the most recent incremental id value
+        page: 1,
+      });
+
+      if (casesResponse.total === 0) {
+        this.logger.debug(`No cases found with incremental id in ${namespace}`);
+        return 0;
+      }
+
+      const mostRecentIncrementalId = casesResponse.saved_objects[0].attributes.incremental_id;
+      this.logger.debug(
+        `getLastAppliedIdForSpace (from cases): Most recent incremental id in ${namespace}: ${mostRecentIncrementalId}`
+      );
+
+      if (mostRecentIncrementalId === undefined || mostRecentIncrementalId === null) {
+        return 0;
+      }
+
+      // TODO: should we really throw here?
+      // There might be gaps because of deleted cases.
+      // if (mostRecentIncrementalId !== casesResponse.total) {
+      //   throw new Error('Mismatch between incremental id and case count');
+      // }
+      return mostRecentIncrementalId;
+    } catch (error) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  /**
+   * Increments the case ids for the given cases.
+   * @param casesWithoutIncrementalId The cases we want to apply IDs to
+   * @param maxDurationMs A maximum timeout to run this task for
```

Comment:
How does this line up with the task manager?

Reply:
This piece of code is not needed anymore, I'll clean that up. I put it there before I learned that the task will be cancelled at intervals as well

Reply:
see [c267ece](https://github.com/elastic/kibana/pull/222874/commits/c267ece31201ba5b665f0b00fa1a1bf438be5600)